src/recommendations/src/recommendations-service/app.py

```diff
@@ -159,12 +173,19 @@ def get_products(feature, user_id, current_item_id, num_results, campaign_arn_pa
     else:
         # Fallback to default behavior of checking for campaign ARN parameter and
         # then the default product resolver.
-        values = get_parameter_values([ campaign_arn_param_name, filter_purchased_param_name ])
+        values = get_parameter_values([default_campaign_arn_param_name, default_filter_arn_param_name])
 
         campaign_arn = values[0]
         filter_arn = values[1]
 
         if campaign_arn and (user_id or not user_reqd_for_campaign):
+
+            logger.info(f"get_products: Supplied campaign: {campaign_arn} (from {default_campaign_arn_param_name}) Supplied filter: {filter_arn} (from {default_filter_arn_param_name}) Supplied user: {user_id}")
+
+            if filter_arn is not None and (user_id is None or len(user_id.strip()) == 0):
+                logger.warning(f"userID is required if using filter - filter Arn: {filter_arn} - spoofing user to 0")
+                user_id = "0"
```

Comment:
UserId is always required for User-Personalization and Personalized-Ranking. For SIMS, it is only required when there is a filter **and** the filter references the users dataset. Since none of the filters in the RDS use the users dataset in their expression, a userId should not required.